x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java

```diff
@@ -500,48 +355,167 @@ private GetAliasesRequest createGetAliasesRequest(FieldCapabilitiesResponse resp
 
     public static List<EsIndex> separateMappings(DataTypeRegistry typeRegistry, String javaRegex, String[] indexNames,
             Map<String, Map<String, FieldCapabilities>> fieldCaps, ImmutableOpenMap<String, List<AliasMetadata>> aliases) {
-        return buildIndices(typeRegistry, indexNames, javaRegex, fieldCaps, aliases, Function.identity(), (s, cap) -> null);
+        return buildIndices(typeRegistry, indexNames, javaRegex, fieldCaps, aliases, Function.identity(), false);
+    }
+
+    private static Map<String, QlFieldCapabilities> buildQlFieldCaps(String[] indexNames,
+            Map<String, Map<String, FieldCapabilities>> fieldCaps, ImmutableOpenMap<String, List<AliasMetadata>> indexToAliases,
+            boolean isMergedMapping) {
```

Comment:
why is merged mapping needed? Shouldn't the logic handle merging anyway (because of aliases)?

Reply:
Validation is needed in two cases:
- when merged mappings are needed (SYS COLUMNS TABLE LIKE and SHOW COLUMNS) and
- when aliases are needed (SYS COLUMNS with wildcard pattern).

When there are no aliases => either we need separate mappings with no aliases or we need merged mappings. But, merged mappings need validation while separate mappings with no aliases doesn't need validation. To distinguish between these two cases when there are no aliases, I added the boolean value.

Reply:
That means for `SYS TABLES` aliasing is not needed explicitly. Do I understand correctly that merging in this case means there's only one result vs one per index?

Reply:
If you meant `SYS COLUMNS TABLE LIKE` then yes, this is a merging operation where the name of the merger is the pattern used in `LIKE` and the columns belong to this one "entity". In this case, the "conflicting" fields are removed from the result.